app/controllers/notifications_controller.rb

```diff
@@ -37,18 +37,28 @@ def index
                        seen_notification_id: current_user.seen_notification_id)
     else
       offset = params[:offset].to_i
-
-      notifications = Notification.where(user_id: user.id)
-        .visible
-        .includes(:topic)
-        .order(created_at: :desc)
-
+      if params[:filter].present? && params[:filter] == "read"
+        notifications = Notification.where(user_id: user.id, read: true)
+          .visible
+          .includes(:topic)
+          .order(created_at: :desc)
+      elsif params[:filter].present? && params[:filter] == "unread"
+        notifications = Notification.where(user_id: user.id, read: false)
+          .visible
+          .includes(:topic)
+          .order(created_at: :desc)
+      else
+        notifications = Notification.where(user_id: user.id)
+          .visible
+          .includes(:topic)
+          .order(created_at: :desc)
+      end
```

Comment:
I think we can reduce some repetition here by doing something like

```suggestion
      notifications = Notification.visible
                     .where(user_id: user.id)
                     .includes(:topic)
                     .order(created_at: :desc)
     
     if params[:filter] == "read"
       notifications = notifications.where(read: true)
     elsif params[:filter] == "unread"
       notifications = notifications.where(read: false)
     end
```